libcoap/tests/common/mod.rs

```diff
@@ -51,34 +57,48 @@ pub(crate) fn spawn_test_server<F: FnOnce(CoapContext<'static>) -> CoapContext<'
     let ready_condition = Arc::new((Mutex::new(false), Condvar::new()));
     let ready_condition2 = Arc::clone(&ready_condition);
 
-    let server_handle = std::thread::spawn(move || {
-        let (ready_var, ready_cond) = &*ready_condition2;
-        run_test_server(|context| {
-            let context = context_configurator(context);
-            let mut ready_var = ready_var.lock().expect("ready condition mutex is poisoned");
-            *ready_var = true;
-            ready_cond.notify_all();
-            context
-        });
-    });
+    let server_handle = std::thread::Builder::new()
+        .name(String::from("test server"))
+        .spawn(move || {
+            let (ready_var, ready_cond) = &*ready_condition2;
+            run_test_server(|context| {
+                let context = context_configurator(context);
+                let mut ready_var = ready_var.lock().expect("ready condition mutex is poisoned");
+                *ready_var = true;
+                ready_cond.notify_all();
+                context
+            });
+        })
+        .expect("unable to spawn test server thread");
 
     let (ready_var, ready_cond) = &*ready_condition;
-    drop(
-        ready_cond
-            .wait_while(ready_var.lock().expect("ready condition mutex is poisoned"), |ready| {
-                !*ready
-            })
-            .expect("ready condition mutex is poisoned"),
-    );
+    {
+        let (_guard, timeout_result) = ready_cond
+            .wait_timeout_while(
+                ready_var.lock().expect("ready condition mutex is poisoned"),
+                Duration::from_secs(10),
+                |ready| !*ready,
+            )
+            .expect("ready condition mutex is poisoned");
+        if timeout_result.timed_out() && server_handle.is_finished() {
+            if let Err(e) = server_handle.join() {
+                std::panic::resume_unwind(e);
+            } else {
+                panic!("Test server thread is dead and has not reported readiness after 10 seconds, but has also not panicked.")
+            }
+        } else if timeout_result.timed_out() {
+            panic!("Test server thread has not reported readiness after 10 seconds, but has also not died (deadlock?).")
+        }
```

Comment:
This is more a matter of personal taste, but I think you could get rid of some of the nesting/elses here which I find a bit more readable. But the original version is also fine.


```suggestion
            if let Err(e) = server_handle.join() {
                std::panic::resume_unwind(e);
            }
            
            panic!("Test server thread is dead and has not reported readiness after 10 seconds, but has also not panicked.")
        }
        
        if timeout_result.timed_out() {
            panic!("Test server thread has not reported readiness after 10 seconds, but has also not died (deadlock?).")
        }
```